pybossa/repositories/task_repository.py

```diff
@@ -544,3 +545,26 @@ def _get_redundancy_update_msg(self, project, n_answers, conditions, params, tas
             task_expiration=task_expiration, **params)).fetchall()
         tasks_not_updated = '\n'.join([str(task.id) for task in tasks])
         return tasks_not_updated
+
+
+    def bulk_update(self, payload):
+        """
+        use sqlalchemy case clause to update db rows in bulk
+        construct payload in the form {task_id: priority} as
+        sqlalchemy case works passing payload in the form
+        WHEN task_id THEN priority
+        https://stackoverflow.com/questions/54365873/sqlalchemy-update-multiple-rows-in-one-transaction
+        """
+
+        if not payload:
+            return
+
+        formatted_payload = {data["id"]: data["priority_0"] for data in payload}
```

Comment:
Would it break is the API payload does not have "id" or "priority_0" field?

Reply:
yes. at this point `id` is required and `priority_0` as well. however, in future, when supporting other attributes to be updated in bulk, `priority_0` won't be required. need to build a logic so that required and option attributes to be present. will make code changes to handle this. Thanks!